588 adding stratified newcombe@main

Melkiades opened the pull request:
# Pull Request

Fixes #588

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+}
+
+#' @describeIn estimate_proportions Calculates the stratified Wilson confidence
+#'   interval for unequal proportions as described in [Yan and Su (2010)].
```
is it correct with the `[]` around Yan and Su reference?
Note that if you would like to cite papers you can consider using `Rdpack`, see https://github.com/insightsengineering/hermes as example where it is used

Melkiades replied:
Thanks for the note, I was unsure how to do it, so I put that placeholder. The core-team suggested me to take inspiration from [this](https://github.com/imbs-hl/ranger/blob/5f71872d7b552fd2cf652daab92416f52976df86/R/ranger.R#L189)

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+#'
+#' @param strata (`factor`)\cr
+#'   with one level per stratum and same length as `rsp`.
+#' @param weights weights for each level of the strata. If missing, they are
```
add class in parentheses

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+#' @param weights weights for each level of the strata. If missing, they are
+#'   estimated using the iterative algorithm proposed in [Yan and Su (2010)]
+#'   that minimizes the weighted squared length of the confidence interval.
+#' @param max_nit maximum number of iterations for the iterative procedure used
```
here too

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+#'   that minimizes the weighted squared length of the confidence interval.
+#' @param max_nit maximum number of iterations for the iterative procedure used
+#'   to find estimates of optimal weights.
+#' @param correct `logical`\cr
```
```suggestion
#' @param correct (`flag`)\cr
```
flag = logical of length 1, cannot be missing etc.

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+#' )
+#'
+#' @export
+prop_strat_wilson <- function(rsp, strata, weights = NULL, conf_level = 0.95, max_nit = NULL, correct = FALSE) {
```
maybe nicer with line breaks after each arg.

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+  # Checking the inputs
+  checkmate::assert_logical(rsp, any.missing = FALSE)
+  checkmate::assert_factor(strata, len = length(rsp))
+  tern:::assert_proportion_value(conf_level)
```
```suggestion
  assert_proportion_value(conf_level)
```
and search for others

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+prop_strat_wilson <- function(rsp, strata, weights = NULL, conf_level = 0.95, max_nit = NULL, correct = FALSE) {
+
+  # Checking the inputs
+  checkmate::assert_logical(rsp, any.missing = FALSE)
```
(would recommend to just import whole `checkmate` package and then avoid all `checkmate::` in package)

Melkiades replied:
This was one of my first discussions with the core-team. I was suggesting using `@importFrom` with the specification of each function to load instead of `::` but they think it is easier to search&find this way and it does not impose loading large namespaces and their consequential conflicts. I personally agree now, even if I do not find it too elegant to add `::` everywhere (above all when dealing with `ggplot2` or `dplyr`).  Still if you see the problems that the `filter` function may cause (namespace conflict between, I think `dplyr` and `stats`), it makes perfect sense to me to impose the standard to add `<package>::`

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+  tern:::assert_proportion_value(conf_level)
+
+  tbl <- table(rsp, strata)
+  n_ws <- ncol(tbl) # Number of centers
```
```suggestion
  n_strata <- ncol(tbl)
```
better more explicit variable name and then no need for comments

Melkiades replied:
I think the comment is important to understand where it comes from that number (I just swapped the two things now, so the variable name is clearer)

danielinteractive replied:
I think it is clear when you say `n_strata` that it is the number of strata

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+  tbl <- table(rsp, strata)
+  n_ws <- ncol(tbl) # Number of centers
+
+  # Checking the weights and maximum number of iterations
```
```suggestion
  # Checking the weights and maximum number of iterations.
```
(sentence style for comments)

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+
+  # Iterative setting of weights if they were not set externally
+  if (do_iter) {
+    ws_old <- weights
+    while (it < max_nit) {
+      ws_new_t <- (1 + strata_qnorm^2 / ns)^2 / (vars + strata_qnorm^2 / (4 * ns^2))
+      ws_new <- ws_new_t / sum(ws_new_t)
+      strata_qnorm <- strata_normal_quantile(vars, ws_new, conf_level)
+      if (sum(abs(ws_new - ws_old)) < 0.001) break
+      ws_old <- ws_new
```
please factor out as helper function (also to reduce length of this function)

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -137,47 +265,97 @@ prop_jeffreys <- function(rsp,
+                           "wilson", "wilsonc", "strat_wilson", "strat_wilsonc",
+                           "agresti-coull", "jeffreys"
                          ),
+                         variables = list(strata = NULL, weights = NULL, max_nit = 10),
```
arg name for `max_nit` could be more explicit, plus check whether 10 is high enough as default

Melkiades replied:
I described it in the new helper function

danielinteractive commented on tests/testthat/test-estimate_proportion.R:
```diff
@@ -11,6 +11,44 @@ testthat::test_that("prop_wilson returns right result", {
+  n_ws <- ncol(table_strata) # Number of weights
+
+  expected <- list(
+    conf.int = c(lower = 0.4073299, upper = 0.5647475),
+    weights = c(0.2074199, 0.1776464, 0.1915610, 0.1604678, 0.1351096, 0.1277952)
```
are these from SAS / other software? or snapshot ?

Melkiades replied:
snapshots

danielinteractive replied:
can you try to compare with SAS?

Melkiades replied:
I devised an issue for this

danielinteractive commented on tests/testthat/test-estimate_proportion.R:
```diff
@@ -11,6 +11,44 @@ testthat::test_that("prop_wilson returns right result", {
+
+  testthat::expect_equal(expected, result, tolerance = 1e-5)
+
+  # Test without estimating weights (all equal here)
```
please separate into 2-3 `test_that()` calls (simplifies debugging etc)

danielinteractive reviewed (commented):
thx @Melkiades , good start! please see inside

Melkiades commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
 #'   [stats::binom.test()]. Also referred to as the `exact` method.
 #' @inheritParams argument_convention
-#' @export
-#' @order 2
```
Checked! It was already in the right order, that is why I took it out

Melkiades commented on R/estimate_proportion.R:
```diff
@@ -32,14 +32,142 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
 }
 
 
+# help function for prop_strat_wilson
```
I will do it. Still, I think these helper functions are used only locally and therefore should be internal. Is it fine if I add `@keywords internal`?

danielinteractive replied:
yep

danielinteractive commented on R/prop_diff.R:
```diff
@@ -243,9 +243,9 @@ prop_diff_cmh <- function(rsp,
-  grp <- as_factor_keep_attributes(grp)
-  strata <- as_factor_keep_attributes(strata)
-  check_diff_prop_ci(
+  grp <- tern:::as_factor_keep_attributes(grp)
+  strata <- tern:::as_factor_keep_attributes(strata)
+  tern:::check_diff_prop_ci(
```
don't need the `tern:::`

Melkiades commented:
Now, it should be complete @danielinteractive. I added the option for getting the `weights` also from the heuristic given by Yan and Su (2010). I also fixed the `NaN` problem. Let me know what you think ^^

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+#'   with one level per stratum and same length as `rsp`.
+#' @param weights (`vector` of `numeric`) \cr
+#'   weights for each level of the strata. If missing, they are
+#'   estimated using the iterative algorithm proposed in (`Yan` and `Su` 2010)
```
same here

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+#'   weights for each level of the strata. If missing, they are
+#'   estimated using the iterative algorithm proposed in (`Yan` and `Su` 2010)
+#'   that minimizes the weighted squared length of the confidence interval.
+#' @param max_nit (`integer`) \cr
```
same here with name

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+#'   stringsAsFactors = TRUE
+#' )
+#' strata <- interaction(strata_data)
+#' n_strata <- ncol(table(rsp, strata)) # Number of weights or centers
```
```suggestion
#' n_strata <- ncol(table(rsp, strata)) # Number of strata
```

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+#'
+#' prop_strat_wilson(
+#'   rsp = rsp, strata = strata,
+#'   weights = rep(1 / n_strata, n_strata), # Not automatic setting of weights
```
I would move the comment above this example

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+#' )
+#' @references
+#' \itemize{
+#'   \item `Yan, Xin, and Xiao Gang Su. 2010. “Stratified Wilson and Newcombe Confidence Intervals for Multiple Binomial Proportions.” Statistics in Biopharmaceutical Research 2 (3): 329–35.`
```
also here I would not put this as code formatting. rather add things to wordlist if needed.

Melkiades replied:
The problem here is that adding everything to the `WORDLIST` file can be tedious as there are around 6-7 words to add every time we add a citation. I thought about something to add such as `# nolint` but @cicdguy told me there is nothing for that and to file an issue. When I did that (here: [spelling#68](https://github.com/ropensci/spelling/issues/68)), I realized that having it as "code" is a good workaround. I think until we do not use `Rdpack` this has to wait

cicdguy replied:
I wouldn't consider adding 6-7 words to the WORDLIST a huge efficiency loss, as there is a function that does this automatically - spelling::update_wordlist(confirm = T).

I think adding that as a pre-commit step is useful, that way you wouldn't have to worry about updating it every time you add new bib references.

Melkiades replied:
I see, very useful! But still, I think it is not too elegant to have a very long `WORDLIST` file that is kind of stacking up without any "cleaning" process. I would prefer using `Rdpack` to have precise references as it is done in `Latex`

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+                              conf_level = 0.95,
+                              max_nit = NULL,
+                              correct = FALSE) {
+
```
```suggestion
```

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+                              max_nit = NULL,
+                              correct = FALSE) {
+
+  # Checking the inputs
```
```suggestion
```
generally, avoid comments when they don't add value

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+  assert_proportion_value(conf_level)
+
+  tbl <- table(rsp, strata)
+  n_strata <- ncol(tbl) # Number of centers and weights
```
```suggestion
  n_strata <- ncol(tbl)
```

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+  ests <- xs / ns
+  vars <- ests * (1 - ests) / ns
+
+  # Estimating quantile for normal distribution large number limit approximation
```
```suggestion
```
etc

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+  # Estimating quantile for normal distribution large number limit approximation
+  strata_qnorm <- strata_normal_quantile(vars, weights, conf_level)
+
+  # Iterative setting of weights if they were not set externally
+  if (do_iter) {
+    ws_new <- update_weights_strat_wilson(vars, strata_qnorm, weights, ns, max_nit, conf_level)$weights
+  } else {
+    ws_new <- weights
+  }
```
```suggestion
  weights_new <- if (do_iter) {
    update_weights_strat_wilson(vars, strata_qnorm, weights, ns, max_nit, conf_level)$weights
  } else {
    weights
  }
```

Melkiades replied:
nice!

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+  lower <- sum(ws_new * lower_by_strata)
+  upper <- sum(ws_new * upper_by_strata)
+
+  # Return values
+  if (do_iter) {
+    list(
+      conf.int = c(
+        lower = lower,
+        upper = upper
+      ),
+      weights = ws_new
+    )
+  } else {
+    list(
+      conf.int = c(
+        lower = lower,
+        upper = upper
+      )
+    )
+  }
```
first create `conf_int` (not with `.` in the name) and then reuse twice

Melkiades replied:
it makes perfect sense. I have just a curiosity: I noticed other instances where `conf.int` is used in `tern`, and it is associated with `summary`, but I do not see the argument in `stats::summary`. Is it meant to be used by [broom::tidy()](https://mpn.metworx.com/packages/broom/0.7.4/reference/tidy.summary.lm.html)?

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -136,48 +354,100 @@ prop_jeffreys <- function(rsp,
+#'
+#' @param df (`logical` vector or `data.frame`)\cr
+#'   if only logical indicates whether each subject is a responder or not.
+#'   `TRUE` represents a successful outcome. If a `data.frame` is provided,
```
if it is a data.frame what is the response column to be called? I would add that here

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -136,48 +354,100 @@ prop_jeffreys <- function(rsp,
 #'   successful outcomes; one of `waldcc`, `wald`, `clopper-pearson`, `wilson`,
-#'   `agresti-coull` or `jeffreys`.
+#'   `wilsonc`, `strat_wilson`, `strat_wilsonc`, `agresti-coull` or `jeffreys`.
+#' @inheritParams rtables::
```
something missing?

Melkiades replied:
Yes, this was weird. I fixed it by adding `argument_convention` to the grouping documentation as it is done in `prop_diff`

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -136,48 +354,100 @@ prop_jeffreys <- function(rsp,
-#' @order 3
 #' @examples
+#'
+#' # Case with only vector of logical
```
```suggestion
#' # Case with only logical vector.
```

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -136,48 +354,100 @@ prop_jeffreys <- function(rsp,
+#'
+#' s_proportion(
+#'   df = dta,
+#'   .var = "rsp",
```
ah good so we can choose - but better for the user to read that in argument description for `df` above already

danielinteractive commented on R/estimate_proportion.R:
```diff
@@ -136,48 +354,100 @@ prop_jeffreys <- function(rsp,
+    stop("To use stratified methods you need to specify the strata variables.")
+  }
+  if (missing(.var)) {
+    rsp <- as.logical(df)
```
this looks risky. what if a data frame with lots of columns is provided?
1) please make this here more defensive
2) please add edge cases in unit tests that catch this

Melkiades replied:
I am unsure how to catch this possibility. Can I ask you for an example? I upload a tentative fix with test for atomic vector

danielinteractive replied:
OK that looks better, maybe it is ok like this

danielinteractive commented on R/prop_diff.R:
```diff
@@ -287,17 +290,140 @@ prop_diff_cmh <- function(rsp,
+#' @describeIn prop_diff Calculates the stratified Newcombe confidence interval
+#'   and difference in response rates between the experimental treatment group
+#'   and the control treatment group, adjusted for stratification factors. This
+#'   implementation follows closely the one proposed by `Yan` and `Su` (2010).
```
same here

danielinteractive commented on R/prop_diff.R:
```diff
@@ -287,17 +290,140 @@ prop_diff_cmh <- function(rsp,
+#'
+#' @references
+#' \itemize{
+#'   \item `Yan, Xin, and Xiao Gang Su. 2010. “Stratified Wilson and Newcombe Confidence Intervals for Multiple Binomial Proportions.” Statistics in Biopharmaceutical Research 2 (3): 329–35.`
```
and here

danielinteractive commented on R/prop_diff.R:
```diff
@@ -287,17 +290,140 @@ prop_diff_cmh <- function(rsp,
+    warning("Less than 5 observations in some strata.")
+  }
+
+  # Splitting
```
```suggestion
```
etc

danielinteractive commented on R/prop_diff.R:
```diff
@@ -287,17 +290,140 @@ prop_diff_cmh <- function(rsp,
+  strata_by_grp <- split(strata, f = grp)
+
+  # Finding the weights
+  if (identical(weights_method, "cmh")) {
+    weights <- prop_diff_cmh(rsp = rsp, grp = grp, strata = strata)$weights
+  } else if (identical(weights_method, "wilson_h")) {
+    weights <- prop_strat_wilson(rsp, strata, conf_level = conf_level, correct = correct)$weights
+  }
```
```suggestion
  weights <- if (identical(weights_method, "cmh")) {
    prop_diff_cmh(rsp = rsp, grp = grp, strata = strata)$weights
  } else if (identical(weights_method, "wilson_h")) {
    prop_strat_wilson(rsp, strata, conf_level = conf_level, correct = correct)$weights
  }
```

danielinteractive commented on R/prop_diff.R:
```diff
@@ -287,17 +290,140 @@ prop_diff_cmh <- function(rsp,
+    weights <- prop_strat_wilson(rsp, strata, conf_level = conf_level, correct = correct)$weights
+  }
+
+  # Calculating L and U per group
```
```suggestion
  # Calculating lower (`l`) and upper (`u`) confidence bounds per group.
```

danielinteractive commented on R/prop_diff.R:
```diff
@@ -287,17 +290,140 @@ prop_diff_cmh <- function(rsp,
+
+  ci_ref <- strat_wilson_by_grp[[1]]
+  ci_trt <- strat_wilson_by_grp[[2]]
+  l1 <- array(ci_ref$conf.int[1]) # array() loses the names
```
better 
```suggestion
  l_ref <- as.numeric(ci_ref$conf.int[1])
```
etc. below - better name and `as.numeric()` does the same but is not confusing that there might be multiple dimensions

Melkiades replied:
that is way better, should I also add `l_trt` for the other values (i.e. `l2` and `*2`)?

Melkiades replied:
I did it for all in the function. I did not extend this to `cmh`

danielinteractive replied:
perfect thx

danielinteractive commented on R/prop_diff.R:
```diff
@@ -287,17 +290,140 @@ prop_diff_cmh <- function(rsp,
+  est2 <- sum(weights * p2)
+  diff_est <- est2 - est1
+
+  # Calculating the final L and U
```
```suggestion
```

danielinteractive commented on R/prop_diff.R:
```diff
@@ -322,16 +452,28 @@ prop_diff_cmh <- function(rsp,
                               .ref_group,
                               .in_ref_col,
-                              variables = list(strata = NULL),
+                              variables = list(strata = NULL, weights_method = "cmh"),
```
hm I think passing `weights_method` in `variables`  does not seem the best place, how about just moving it to a normal argument?

danielinteractive commented on R/prop_diff.R:
```diff
@@ -347,21 +489,50 @@ s_proportion_diff <- function(df,
+      checkmate::assert_character(strata_colnames, null.ok = FALSE)
+      strata_vars <- stats::setNames(as.list(strata_colnames), strata_colnames)
+
+      # Checks if strata variables are in a well formatted data frame
```
```suggestion
```

danielinteractive reviewed (commented):
thanks a lot @Melkiades , nice work! Please see comments inside.
(In general I would reduce comments in the code a lot to the real minimum, think more about better variable names and avoid `.` in names)

Melkiades commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+    if (diff_v[length(diff_v)] < tol) break
+    ws_old <- ws_new
+    it <- it + 1
+  }
```
Yes, I agree

Melkiades commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+#'   for references and details.
+#'
+#' @inheritParams prop_strat_wilson
+#' @param vars (`vector` of `numeric`) \cr
```
No, I do not think it changed but I put it this way to raise actually the question if it is better to be more informative on the input format. I think it is good to know what is the detailed intended input. It makes things clearer if you are trying things out. What do you think?

Melkiades replied:
ok, it makes sense! I think for other variables, I could specify the same as `checkmate` checks. For example, what do you think about defining `choice` for the method variable?

danielinteractive replied:
I would just say `string` for that

danielinteractive replied:
we have in the dev guide a table actually that defines what to use

Melkiades replied:
Found it: [https://nest.pages.roche.com/docs/dev-guide/coding/documentation/](https://nest.pages.roche.com/docs/dev-guide/coding/documentation/). Thanks 👍

danielinteractive replied:
perfect - thanks Davide

Melkiades commented on R/estimate_proportion.R:
```diff
@@ -31,15 +31,233 @@ prop_wilson <- function(rsp, conf_level, correct = FALSE) {
+  ))
+}
+#' @describeIn estimate_proportions Calculates the stratified Wilson confidence
+#'   interval for unequal proportions as described in (`Yan` and `Su` 2010).
```
`Rdpack` seems the best option to me, but `tern` does not have anything in place for this. I think I will go for the first option and spawn a disccusion/issue about implementing option 2. Is it fine with you? Do you use `Rdpack` already? So, I can link it as an example

danielinteractive commented:
@Melkiades please request re-review once ready, thanks

Melkiades commented:
> @Melkiades please request re-review once ready, thanks

I am just not sure about the `missing(.var)` alternative. I need to find a case in which this breaks. ATM, there is a `test_atomic_vector(df)` in place. I am also working on adding `Rdpack`, if @cicdguy is fine with it, but it will be another issue

danielinteractive commented on R/prop_diff.R:
```diff
@@ -389,35 +387,34 @@ prop_diff_strat_nc <- function(rsp,
+  u_trt <- as.numeric(ci_trt$conf_int[2])
 
-  # Estimating the diff and n1, n2 (it allows different weights to be used)
+  # Estimating the diff and n_ref, n2 (it allows different weights to be used)
```
```suggestion
  # Estimating the diff and n_ref, n_trt (it allows different weights to be used)
```

danielinteractive commented on tests/testthat/test-estimate_proportion.R:
```diff
@@ -74,13 +74,17 @@ testthat::test_that("update_weights_strat_wilson convergence test", {
   ni <- 1000 # Maximum number of allowed iterations
   tol <- 0.0001 # Tolerance for convergence
-  sq <- strata_normal_quantile(vs, ws, cl) # Initial quantiles
+  sq <- tern:::strata_normal_quantile(vs, ws, cl) # Initial quantiles
```
this should work without the `tern:::` prefix? In tests, internal functions are available.

Melkiades replied:
My apologies, I did it on Friday, forgot to take it out

danielinteractive reviewed (approved):
thanks @Melkiades , this looks much better, good to merge after resolving the last inside comments. Good then to follow up with 1) SAS comparison, 2) we need to expose this option especially the stratified Newcombe CI in teal.modules.clinical, 3) (less time sensitive) is the `Rdpack` thing.
